src/software/gui/robot_diagnostics/widgets/robot_diagnostics_gui.cpp

```diff
@@ -33,24 +41,162 @@ void RobotDiagnosticsGUI::onChickerStateChanged(double chicker_power,
                                                 ChargeMode charge_mode,
                                                 ChickMode chick_mode)
 {
-    // TODO (Issue #1420): push primitive set msg to buffer
+    if (chick_mode == ChickMode::CHIP)
+    {
+        chip_pressed = true;
+    }
+    else if (chick_mode == ChickMode::KICK)
+    {
+        kick_pressed = true;
+    }
+    auto direct_control_primitive = createDirectControlPrimitiveFromUI();
+    pushPrimitiveSetToBuffer(std::move(direct_control_primitive));
 }
 
 void RobotDiagnosticsGUI::onDirectVelocityPowerChanged(
     double direct_per_wheel_power, DirectVelocityMode direct_velocity_mode)
 {
-    // TODO (Issue #1420): push primitive set msg to buffer
+    auto direct_control_primitive = createDirectControlPrimitiveFromUI();
+    pushPrimitiveSetToBuffer(std::move(direct_control_primitive));
 }
 
 void RobotDiagnosticsGUI::onDirectPerWheelPowerChanged(
     double direct_per_wheel_power, DirectPerWheelMode direct_per_wheel_mode)
 {
-    // TODO (Issue #1420): push primitive set msg to buffer
+    auto direct_control_primitive = createDirectControlPrimitiveFromUI();
+    pushPrimitiveSetToBuffer(std::move(direct_control_primitive));
 }
 
 void RobotDiagnosticsGUI::onDribblerPowerChanged(double dribbler_power)
 {
-    // TODO (Issue #1420): push primitive set to buffer
+    auto direct_control_primitive = createDirectControlPrimitiveFromUI();
+    pushPrimitiveSetToBuffer(std::move(direct_control_primitive));
+}
+
+std::unique_ptr<TbotsProto::Primitive>
+RobotDiagnosticsGUI::createDirectControlPrimitiveFromUI()
+{
+    auto direct_control_primitive_msg = std::make_unique<TbotsProto::Primitive>();
+
+    // Uses currently selected tabWidget to decide which wheel control Primitive to make
+    switch (main_widget->tabWidget->currentIndex())
+    {
+        case static_cast<int>(WheelControlTab::DIRECT_PER_WHEEL):
```

Comment:
It would be cleaner to have these cases in helper functions instead of having a super long function like this.